File: Command/DotNet/Command/executors/Fan.cs
namespace CommandPattern.Executors;


public class CeilingFan
{
    public enum Speeds
    {
        HIGH = 3,
        MEDIUM = 2,
        LOW = 1,
        OFF = 0,
    };
    public string Name { get; set; }
    public Speeds Speed { get; set; } = Speeds.OFF;

    public CeilingFan(string name)
    {
        this.Name = name;
    }

    public void High()
    {
        this.Speed = Speeds.HIGH;
    }

    public void Medium()
    {
        this.Speed = Speeds.MEDIUM;
    }

    public void Low()
    {
        this.Speed = Speeds.LOW;
    }

    public void Off()
    {
        this.Speed = Speeds.OFF;
    }
}

File: Command/Python/src/executors/fan.py
from enum import Enum


class Fan:
    class Speeds(Enum):
        HIGH = 3
        MEDIUM = 2
        LOW = 1
        OFF = 0

    def __init__(self, name: str) -> None:
        self._name = f"{name} fan"
        self.speed = self.Speeds.HIGH

    @property
    def name(self) -> str:
        return self._name

    @name.setter
    def name(self) -> None:
        raise AttributeError("'name' is read only")

    def high(self) -> None:
        self.speed = self.Speeds.HIGH

    def medium(self) -> None:
        self.speed = self.Speeds.MEDIUM

    def low(self) -> None:
        self.speed = self.Speeds.LOW

    def off(self) -> None:
        self.speed = self.Speeds.OFF